test/e2e/network_test.go

```diff
@@ -0,0 +1,81 @@
+package e2e
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/retry"
+	wmc "github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig"
+	"github.com/openshift/windows-machine-config-operator/pkg/controller/windowsmachineconfig/nodeconfig"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const (
+	// hybridOverlayMac is an annotation applied by the hybrid-overlay
+	hybridOverlayMac = "k8s.ovn.org/hybrid-overlay-distributed-router-gateway-mac"
+)
+
+// testNetwork runs all the cluster and node network tests
+func testNetwork(t *testing.T) {
+	t.Run("Hybrid overlay annotations", testHybridOverlayAnnotations)
+	t.Run("OpenShift HNS networks", testHNSNetworksCreated)
+}
+
+// testHybridOverlayAnnotations tests that the correct annotations have been added to the bootstrapped node
+func testHybridOverlayAnnotations(t *testing.T) {
+	testCtx, err := NewTestContext(t)
+	require.NoError(t, err)
+	defer testCtx.cleanup()
+
+	nodes, err := testCtx.kubeclient.CoreV1().Nodes().List(
+		metav1.ListOptions{LabelSelector: wmc.WindowsOSLabel})
+	require.NoError(t, err, "unable to get Windows nodes")
+	for _, node := range nodes.Items {
+		assert.NoError(t, testCtx.waitForNodeAnnotation(node.GetName(), nodeconfig.HybridOverlaySubnet))
+		assert.NoError(t, testCtx.waitForNodeAnnotation(node.GetName(), hybridOverlayMac))
+	}
+}
+
+// waitForNodeAnnotation check if the node object has the given annotation and waits for retry.Interval seconds and
+// returns an error if the annotation does not appear in that time frame.
+func (tc *testContext) waitForNodeAnnotation(nodeName, annotation string) error {
```

Comment:
This code is being duplicated in the functionality and test cases (waitForHybridOverlayAnnotation). 
You can move this to a shared package and call this within the waitForHybridOverlayAnnotation function.

Reply:
Aren't we waiting for the annotation in the main code? Shouldn't we just get the nodeObject and check for annotation here?

Reply:
>This code is being duplicated in the functionality and test cases (waitForHybridOverlayAnnotation).
You can move this to a shared package and call this within the waitForHybridOverlayAnnotation function.

I do not want to share main and test code.

>Aren't we waiting for the annotation in the main code? Shouldn't we just get the nodeObject and check for annotation here?

We are only waiting for one of the annotations in the main code and I did not want to add extra code where we wait for one and not for the other.

Reply:
Same as above

Reply:
Given https://github.com/openshift/windows-machine-config-operator/pull/21#discussion_r405539281, I will not wait for the annotations anymore.